src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -700,6 +701,11 @@ def filter_out_esm_packages(self, packages, package_versions):
             self.status_handler.add_error_to_status("{0} patches requires Ubuntu Pro for Infrastructure with Extended Security Maintenance".format(len(esm_packages)), Constants.PatchOperationErrorCodes.UA_ESM_REQUIRED) # Set the error status with the esm_package details. Will be used in portal.
             self.telemetry_writer.write_event("Filter esm packages [EsmPackagesCount={0}]".format(len(esm_packages)), Constants.TelemetryEventLevel.Informational)
 
-        self.composite_logger.log_debug("Filter esm packages : [TotalPackagesCount={0}][EsmPackagesCount={1}]".format(len(packages), len(esm_packages)))
+        # Log the Esm packages that will be patched in case of an Ubuntu Pro attached VM.
+        security_esm_update_query_success, security_esm_updates, security_esm_updates_versions = self.get_security_esm_updates()
```

Comment:
Can we avoid calling self.get_security_esm_updates()?

Currently self.get_security_esm_updates() is called from set_security_esm_package_status() and here. We can call self.get_security_esm_updates() only once and use the list for both: logging and set_security_esm_package_status().

Reply:
For patching, This method is called first and then followed by the "set_security_esm_package_status". 
For Assessment, Only "set_security_esm_package_status" is called.
So, its difficult to put it in one place, as we still have to maintain if the query was success and other details. Hence, have to call placed in this.